packages/x-adapter/README.md

```diff
@@ -1,13 +1,218 @@
+  requestMapper({ query, start, rows }) {
+    return {
+      q: query,
+      limit: rows,
+      offset: start
+      // ... any field you need to send for your request
+    };
+  },
+  responseMapper({ results }) {
+    return {
+      results: results,
+      totalResults: results.length
+      // ... any field you need to map from the response
+    };
+  }
+};
+```
+
+#### Using a factory function to configure your adapter
+
+To ease this configuration, you can also use the `x-adapter` ’s `endpointAdapterFactory` function
+that defines how a request will be triggered and resolved with the given `EndpointAdapterOptions`.
+
+```tsx
+import { endpointAdapterFactory } from '@empathyco/x-adapter';
+import { SearchRequest, SearchResponse } from '@empathyco/x-types';
+
+export const searchEndpointAdapter = endpointAdapterFactory<SearchRequest, SearchResponse>({
+  endpoint: 'https://{API_ENDPOINT}',
+  defaultRequestOptions: {
+    properties: {
+      headers: { Authorization: `${YOUR_AUTH}` }
+      // ... any options your API may need
+    }
+  },
+  // we enhance you to declare your resquest/response mappers ouside this function to improve its legibility
+  requestMapper: searchRequestMapper,
+  responseMapper: searchResponseMapper
+});
+```
+
+### Implement your own adapter using schemas
+
+The `x-adapter` library includes a `schemaMapperFactory` function that creates a mapper function to
+execute the schema you provide. You just need to create your request and response schemas to define
+the structure of your data, and pass them as a parameter to the factory function.
+
+```tsx
+import { schemaMapperFactory } from '@empathyco/x-adapter';
+import { SearchRequest } from '@empathyco/x-types';
+import { searchRequestSchema } from '../../schemas/requests/search-request.schema';
+
+// Import and use already existing types from your API source, or declare your own based on the response you get.
+interface ApiRequestModel {
+  q: string;
+  limit?: number;
+  offset?: number;
+}
+interface ApiResponseModel {
+  items: T[];
+  limit?: number;
+  offset?: number;
+  total: number;
+}
+
+// Maps both the request and the response to connect your model with API you are working with.
+const searchRequestSchema: Schema<SearchRequest, ApiRequestModel> = {
+  q: 'query',
+  limit: 'rows',
+  offset: 'start'
+};
+const searchResponseSchema: Schema<ApiResponseModel, SearchResponse> = {
+  results: 'items',
+  totalResults: 'items.total' // you can use paths to access to an inner property if needed
+};
+
+// The 'schemaMapperFactory' creates a mapper function for a given schema.
+export const searchRequestMapper = schemaMapperFactory<SearchRequest, ApiRequestModel>(
+  searchRequestSchema
+);
+export const searchResponseMapper = schemaMapperFactory<ApiResponseModel, SearchResponse>(
+  searchResponseSchema
+);
+```
+
+#### Create more complex models with Subschemas
+
+For more complex data mapping, the `x-adapter` library also provides a `subSchema` type that
```

Comment:
I wouldn't say that `x-adapter` provides with `subSchemas`.  The idea of subschemas is being able to re-use a schema between different adapters and with different paths. Something like this maybe?. I would also change the example reusing the same schema under different paths in two adapters. 
> When you are creating adapters for different APIs you might find the case that you have to map the same model in different places. To help you with that, schemas allows to use subschemas. To use them you just have to provide with the path of the data to map, and the schema to apply to it.